### PR TITLE
Make tick.c atomic

### LINKED ISSUE #173: uint64_t is returned non-atomically without a guard lock
The tick function is not interrupt safe due to the use of a uint64_t. We should either implement guarding or reduce this to an atomic uint32_t

https://github.com/kanflo/opendps/blob/c1a8ab69d97ee3a329a2283ec88c49150537978c/opendps/tick.c#L30
https://github.com/kanflo/opendps/blob/c1a8ab69d97ee3a329a2283ec88c49150537978c/opendps/tick.c#L67

### PR DESCRIPTION
**Pull request by Xenoamor**
Resolves #173 

I've split the `uint64_t` into two `uint32_t`s so that `delay_ms(uint32_t)` doesn't have to keep disabling interrupts to access the current tick value. This would have thrown off the delay functions accuracy

### CODE REVIEW

File: opendps/tick.c [hunk 1]
```diff
@@ -51,20 +52,24 @@ void systick_init(void)
...
   */
-
 void delay_ms(uint32_t delay)
 {
-    uint64_t start = tick_ms;
-    while (tick_ms < start + delay) ;    
+    uint32_t start = ticks_lower;
+
+    while ((ticks_lower - start) < delay)
+    {}
```
Comment (kanflo) [inline opendps/tick.c]: This will fail when ```ticks_lower``` wraps between lines 57 and 59 right?

**Comment by Xenoamor** [comment]
It looks strange but it is correct. Remember that ticks_lower is at a later
time then start. This quote describes it well:

"The counter goes (e.g. for 8 bits) 0xfc, 0xfd, 0xfe, 0xff, 0x00, 0x01,
0x02, 0x03 etc., and you know that of the two values x and y that you have,
x comes later. If x==0x02 and y==0xfe, the calculation x-y (as an 8-bit
result) will give the correct answer of 4, assuming that subtraction of two
*n*-bit values wraps modulo 2n - which C99 guarantees for subtraction of
unsigned values."

Seen here:
https://stackoverflow.com/questions/2061245/how-to-subtract-two-unsigned-ints-with-wrap-around-or-overflow

Basically this only works if y is fixed and x only increases (including
overflowing)

File: opendps/tick.c [hunk 2]
```diff
@@ -73,6 +78,8 @@ uint64_t get_ticks(void)
   */
 void sys_tick_handler(void)
 {
-    tick_ms++;
+    ticks_lower++;
+    if (ticks_lower == 0) // If an overflow has occured
+        ticks_upper++;
 }
```
Comment (Xenoamor) [inline opendps/tick.c]: This assumes that `get_ticks()` is never called from an interrupt with a higher priority

Comment (kanflo) [reply opendps/tick.c]: Ah, clever! Nice to learn something new every now and then 🎉
